### PR TITLE
Update Terraform workflow

### PR DESCRIPTION
**Pull request by ryanwholey**
Updates the terraform.yml workflow template:
- Bumps check modules to v2 (targeting Terraform Cloud instead of Enterprise)
- Adds a commented  out "workspace" job, which can be uncommented to manage Terraform Cloud workspaces via the workflow

Any approaches that come to mind other than uncommenting? Checking for a "backend" definition could be nice, but they are not technically required to run plans against cloud, and as such have not been implemented across the board (ex [kiva](https://github.com/TakeScoop/kiva/blob/master/terraform/backend.tf)). I'm inclined to start here and develop a more robust solution as needed. 

Once we nail down this template, I'll update the handbook.

### CODE REVIEW

**Comment by bendrucker** [comment]
Setting `if` for the job would probably be better, that would allow for at least a single line to comment out/in to enable.

**Comment by ryanwholey** [comment]
@bendrucker updated a bit, lmk what you think!

File: workflow-templates/terraform.yml [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+    if: false
+    uses: TakeScoop/terraform-cloud-workspace-github-workflow/.github/workflows/terraform-workspace.yml@v1
+    secrets:
+      terraform_token: ${{ secrets.TFC_TOKEN_WORKSPACE_MANAGEMENT }}
+    with:
+      # Set working_directory to the directory containing your Terraform code. Default: './'
+      # working_directory: terraform
+
+      # Each environment entry creates one Terraform workspace, ${name}-${workspace}, where ${name} defaults to the repository name.
+      # environments: |-
```
Comment (bendrucker) [inline workflow-templates/terraform.yml:34]: This should probably be enabled by default

File: workflow-templates/terraform.yml [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
 jobs:
   check:
-    uses: takescoop/terraform-check-modules-workflow/.github/workflows/check.yml@v1
+    uses: takescoop/terraform-check-modules-workflow/.github/workflows/check.yml@v2
     secrets:
-      terraform_token: ${{ secrets.TF_TOKEN_GET_MODULES }}
+      terraform_token: ${{ secrets.TFC_TOKEN_GET_MODULES }}
+  workspace:
+    # Remove this "if" line to enable Terraform workspace management.
+    if: false
```
Comment (bendrucker) [inline workflow-templates/terraform.yml:25]:
```suggestion
    if: false # Remove this line to enable Terraform Cloud workspaces
```

File: workflow-templates/terraform.yml [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
-      terraform_token: ${{ secrets.TF_TOKEN_GET_MODULES }}
+      terraform_token: ${{ secrets.TFC_TOKEN_GET_MODULES }}
+  workspace:
+    if: false # Remove this line to enable Terraform Cloud workspaces
+    uses: TakeScoop/terraform-cloud-workspace-github-workflow/.github/workflows/terraform-workspace.yml@v1
+    secrets:
+      terraform_token: ${{ secrets.TFC_TOKEN_WORKSPACE_MANAGEMENT }}
+    with:
+      # Set working_directory to the directory containing your Terraform code.
+      working_directory: ./
```
Comment (ryanwholey) [inline workflow-templates/terraform.yml:30]:
```suggestion
      working_directory: terraform/
```

If we're going to include environments in the default workflow, I think it probably makes sense to use `terraform/` as the default working directory. Most of our users will be using environments with this working directory set

File: workflow-templates/terraform.yml [hunk 4]
```diff
@@ -1,16 +1,35 @@
...
-      terraform_token: ${{ secrets.TF_TOKEN_GET_MODULES }}
+      terraform_token: ${{ secrets.TFC_TOKEN_GET_MODULES }}
+  workspace:
+    if: false # Remove this line to enable Terraform Cloud workspaces
+    uses: TakeScoop/terraform-cloud-workspace-github-workflow/.github/workflows/terraform-workspace.yml@v1
+    secrets:
+      terraform_token: ${{ secrets.TFC_TOKEN_WORKSPACE_MANAGEMENT }}
+    with:
+      # Set working_directory to the directory containing your Terraform code.
+      working_directory: terraform/
```
Comment (bendrucker) [inline workflow-templates/terraform.yml:30]:
```suggestion
      working_directory: terraform/ # Remove if Terraform configuration is stored in the repo root
```

File: workflow-templates/terraform.properties.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Manage Terraform Resources",
```
Comment (bendrucker) [inline workflow-templates/terraform.properties.json:2]:
```suggestion
  "name": "Terraform",
```

File: workflow-templates/terraform.properties.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  "description": "Manage all Terraform related tasks, including managing Terraform Cloud workspaces and format checking, linting and validating Terraform code",
```
Comment (bendrucker) [inline workflow-templates/terraform.properties.json:3]:
```suggestion
  "description": "Comprehensive workflow for Terraform repositories, including Cloud workspaces and linting/validation.",
```

**Review by bendrucker — APPROVED** [review]

**Pull request merged**